internal/manifests/monolithic/configmap.go

```diff
@@ -152,7 +154,19 @@ func buildTempoConfig(opts Options) ([]byte, error) {
 	tempo := opts.Tempo
 
 	config := tempoConfig{}
+	config.MultitenancyEnabled = tempo.Spec.Multitenancy != nil && tempo.Spec.Multitenancy.Enabled
 	config.Server.HttpListenPort = manifestutils.PortHTTPServer
+	if tempo.Spec.Multitenancy.IsGatewayEnabled() {
+		// all connections to tempo must go via gateway
+		config.Server.HTTPListenAddress = "localhost"
+		config.Server.GRPCListenAddress = "localhost"
+	}
+
+	// The internal server is required because if the gateway is enabled,
+	// the Tempo API will listen on localhost only,
+	// and then Kubernetes cannot reach the health check endpoint.
+	config.InternalServer.Enable = true
```

Comment:
Wondering if we cannot have the internal server always enabled? :D

Reply:
Yes, I think we should always enable it in the `TempoStack` as well.
It is always enabled in `TempoMonolithic`.